### DISCUSSION TITLE
Error when transcoding after updating proxmox (Jellyfin)

### DISCUSSION THREAD

**Discussion by rcv11x** [tteck/Proxmox#1056] [Help Needed]
hi, i have updated proxmox and before jellyfin worked fine and transcoded if necessary, now when i play something it tells me that the media is not compatible, and in the transcoding log i get this error, what could have happened, has it broken something? 

note: jellyfin I have it with lxc and I installed it with the tteck script.

[error_log.txt](https://github.com/tteck/Proxmox/files/10721661/error_log.txt)

**Comment by tteck** [comment]
Recheck FFmpeg path: `/usr/lib/jellyfin-ffmpeg/ffmpeg`

**Reply by rcv11x** [reply]
I checked and it was fine
![Captura de pantalla de 2023-02-13 12-40-45](https://user-images.githubusercontent.com/38696644/218448677-7a370cd1-02db-428e-83a6-58b1a83548e3.png)
![Captura de pantalla de 2023-02-13 12-41-28](https://user-images.githubusercontent.com/38696644/218448748-d0a46232-6355-4f9f-96b4-4e6f509121f9.png)

**Reply by tteck** [reply]
Share your `.conf`
Run command `cat /etc/pve/lxc/<CTID>.conf` in the Proxmox shell

**Reply by rcv11x** [reply]
`
## Jellyfin LXC
#### https%3A//tteck.github.io/Proxmox/
#<a href='https%3A//ko-fi.com/D1D7EP4GF'><img src='https%3A//img.shields.io/badge/%E2%98%95-Buy me a coffee-red' /></a>
arch: amd64
cores: 6
features: nesting=1
hostname: jellyfin
memory: 4096
mp0: almacen1:vm-102-disk-0,mp=/jelly-data,backup=1,size=2000G
net0: name=eth0,bridge=vmbr0,gw=192.168.96.1,hwaddr=22:1A:4A:56:2F:2E,ip=192.168.96.12/24,type=veth
onboot: 1
ostype: ubuntu
parent: snap_01_02_2022
rootfs: vmstore:vm-102-disk-0,size=50G
swap: 512
tags: 192.168.96.12;debian;lxc
lxc.cgroup2.devices.allow: c 226:0 rwm
lxc.cgroup2.devices.allow: c 226:128 rwm
lxc.cgroup2.devices.allow: c 29:0 rwm
lxc.mount.entry: /dev/fb0 dev/fb0 none bind,optional,create=file
lxc.mount.entry: /dev/dri dev/dri none bind,optional,create=dir
lxc.mount.entry: /dev/dri/renderD128 dev/renderD128 none bind,optional,create=file

[bueno]
#antes de instalar certbot y nginx para hacer https
arch: amd64
cores: 4
features: nesting=1
hostname: jellyfin
memory: 4096
mp0: almacen1:vm-102-disk-0,mp=/jelly-data,backup=1,size=500G
net0: name=eth0,bridge=vmbr0,gw=192.168.96.1,hwaddr=22:1A:4A:56:2F:2E,ip=192.168.96.12/24,type=veth
onboot: 1
ostype: ubuntu
rootfs: vmstore:vm-102-disk-0,size=50G
snaptime: 1673959134
swap: 512
tags: 192.168.1.12;debian;lxc
lxc.cgroup2.devices.allow: c 226:0 rwm
lxc.cgroup2.devices.allow: c 226:128 rwm
lxc.cgroup2.devices.allow: c 29:0 rwm
lxc.mount.entry: /dev/fb0 dev/fb0 none bind,optional,create=file
lxc.mount.entry: /dev/dri dev/dri none bind,optional,create=dir
lxc.mount.entry: /dev/dri/renderD128 dev/renderD128 none bind,optional,create=file

[snap_01_02_2022]
#Todo bien antes de instalar smb users y demas
arch: amd64
cores: 6
features: nesting=1
hostname: jellyfin
memory: 4096
mp0: almacen1:vm-102-disk-0,mp=/jelly-data,backup=1,size=2000G
net0: name=eth0,bridge=vmbr0,gw=192.168.96.1,hwaddr=22:1A:4A:56:2F:2E,ip=192.168.96.12/24,type=veth
onboot: 1
ostype: ubuntu
parent: bueno
rootfs: vmstore:vm-102-disk-0,size=50G
snaptime: 1675268097
swap: 512
tags: 192.168.96.12;debian;lxc
lxc.cgroup2.devices.allow: c 226:0 rwm
lxc.cgroup2.devices.allow: c 226:128 rwm
lxc.cgroup2.devices.allow: c 29:0 rwm
lxc.mount.entry: /dev/fb0 dev/fb0 none bind,optional,create=file
lxc.mount.entry: /dev/dri dev/dri none bind,optional,create=dir
lxc.mount.entry: /dev/dri/renderD128 dev/renderD128 none bind,optional,create=file
`

**Reply by tteck** [reply]
Add these two lines above `lxc.cgroup2.devices.allow: c 226:0 rwm`, reboot, and see if that helps
```
lxc.cgroup2.devices.allow: a
lxc.cap.drop:
```

**Reply by rcv11x** [reply]
I have added those 2 lines but it still doesn't work
[error_log.txt](https://github.com/tteck/Proxmox/files/10722392/error_log.txt)


![Captura de pantalla de 2023-02-13 13-59-15](https://user-images.githubusercontent.com/38696644/218464737-cc59d8bb-07c0-4aee-b6fa-18cadca53cdb.png)

**Reply by rcv11x** [reply]
is it possible that with the proxmox update I lost the intel igpu drivers? with the use of your script I installed something similar? is it possible that something is broken or deleted?

**Reply by lightmaster** [reply]
You mention "the Intel igpu drivers", does that mean that the Jellyfin LXC created by the script does support Intel Quicksync for hardware transcoding? I don't see it explicitly mentioned like it is for the Plex script.

**Reply by memphisraynz** [reply]
I'm seeing the same issue. 
This is newly installed JellyFin, enabled Hardware Transcoding, checked the ffmpeg path but get the same frontend error and the same error in the logs
```
[AVHWDeviceContext @ 0x559a1012b0c0] No VA display found for any default device.
Device creation failed: -22.
Failed to set value 'vaapi=va:,driver=iHD,kernel_driver=i915' for option 'init_hw_device': Invalid argument
Error parsing global options: Invalid argument
```
I have plex from your scripts running on the same host and it works. Plex was shutdown while testing JellyFin.
Running the tests shown on the JellyFin page says that I have a valid GPU
![image](https://github.com/tteck/Proxmox/assets/9811720/f2be926a-7f36-4bf1-98c3-dd2b2be257b4)

**Reply by jarjumarvin** [reply]
Having the same issue on my new server.

**Reply by kylehase** [reply]
This appears to be an issue with the group ID assigned to the rendering device which may have changed between Proxmox 7->8.

I'm not sure if this is the ideal fix but it worked for me. From the Jellyfin PVE console:
1. Check the groups to which the Jellyfin user belongs. 
```
root@jellyfin:/dev/dri# groups jellyfin
jellyfin : jellyfin video render
```
2. Check the group owner of the renderer. In my case the group owner is `ssl-cert` but jellyfin is not a part of that group.
 ```bash
root@jellyfin:/dev/dri# ls -l /dev/dri/
total 0
drwxr-xr-x 2 root root           80 Jul 14 22:18 by-path
crw-rw---- 1 root video    226,   0 Jul 14 22:18 card0
crw-rw---- 1 root ssl-cert 226, 128 Jul 14 22:18 renderD128
```
3. Change the rendering device group owner to `render`.
```
root@jellyfin:/dev/dri# chgrp render /dev/dri/renderD128 
root@jellyfin:/dev/dri# ls -l /dev/dri/
total 0
drwxr-xr-x 2 root root         80 Jul 14 22:18 by-path
crw-rw---- 1 root video  226,   0 Jul 14 22:18 card0
crw-rw---- 1 root render 226, 128 Jul 14 22:18 renderD128
```

**Reply by Zenopheus** [reply]
This fixed it for me. Thank you!

**Reply by Stooovie** [reply]
Didn't fix here. Mine also stopped working recently, also LXC, and i915. QSV was working, now I get the OP's error message.
my content of /dev/dri is different though, I only have one entry.

```
root@jelly:~# ls -l /dev/dri/
total 0
---------- 1 root render 0 Nov  6 17:10 renderD128
```

`chgrp render /dev/dri/renderD128` does not fix the issue. What else can I do?

**Reply by nickschot** [reply]
On my (fresh) proxmox 8 setup I could only get it working if I create a Debian (12) CT with the script (defaults otherwise, so privileged container), it didn't work with the default Ubuntu CT. Also had to add the jellyfin user to the `sgx` group, since that's how `renderD128` showed up for me.

**Reply by shaulbarlev** [reply]
I have the same issue- upon host restart the group for renderD128 changes to ssl-cert from inside the jellyfin LXC. chgrp easily fixes this but looks like I have to run this after every reboot of the host.

**Reply by iodize6399** [reply]
I confirm this misbehavior as well. I have to change the container's rendering device's group every time I restart the host.

**Reply by tteck** [reply]
Try this to see if it resolves the issue `adduser $(id -u -n) video && adduser $(id -u -n) render`

**Reply by iodize6399** [reply]
I corrected a syntax error and executed to no avail
```shell
root@jellyfin:~# adduser $(id -u -n) video && adduser $(id -u -n) render
The user `root' is already a member of `video'.
The user `root' is already a member of `render'.
root@jellyfin:~# ls -l /dev/dri/
total 0
drwxr-xr-x 2 root root           80 Jan  1 01:20 by-path
crw-rw---- 1 root video    226,   0 Jan  1 01:20 card0
crw-rw---- 1 root ssl-cert 226, 128 Jan  1 01:20 renderD128
```
```shell
root@jellyfin:~# chgrp render /dev/dri/renderD128
root@jellyfin:~# ls -l /dev/dri/
total 0
drwxr-xr-x 2 root root         80 Jan  1 01:20 by-path
crw-rw---- 1 root video  226,   0 Jan  1 01:20 card0
crw-rw---- 1 root render 226, 128 Jan  1 01:20 renderD128
```

**Reply by tteck** [reply]
Aww... I see. 

```
chown root:render /dev/dri/renderD128
chmod 0660 /dev/dri/renderD128
```

**Reply by iodize6399** [reply]
```shell
root@jellyfin:~# ls -l /dev/dri/
total 0
drwxr-xr-x 2 root root           80 Jan  1 15:07 by-path
crw-rw---- 1 root video    226,   0 Jan  1 15:07 card0
crw-rw---- 1 root ssl-cert 226, 128 Jan  1 15:07 renderD128

root@jellyfin:~# chown root:render /dev/dri/renderD128
chmod 0660 /dev/dri/renderD128
root@jellyfin:~# ls -l /dev/dri/
total 0
drwxr-xr-x 2 root root         80 Jan  1 15:07 by-path
crw-rw---- 1 root video  226,   0 Jan  1 15:07 card0
crw-rw---- 1 root render 226, 128 Jan  1 15:07 renderD128
```
Still doesnot persist after a host reboot.

**Reply by shaulbarlev** [reply]
This is my workaround to solve the renderD128 group changing from render:
Do the following inside the Jellyfin LXC:
`crontab -e`
add this to the end:
`@reboot chgrp render /dev/dri/renderD128`

let me know if this helped
cheers

**Reply by DoTheSneedful** [reply]
Try `sudo adduser jellyfin ssl-cert` then restart the CT

Quite frankly I don't know why /dev/dri/renderD128 has that group but I assume it's related to proxmox setting up the GPU passthrough. Adding the jellyfin user to that group fixed this for me and it survives CT/host restarts without needing to do anything else.

**Comment by tteck** [comment]
Need testers https://github.com/tteck/Proxmox/issues/2523#issuecomment-1955723524
